iaso/api/instances.py

```diff
@@ -314,9 +314,9 @@ def list(self, request):
         # 2. Prepare queryset (common part between searches and exports)
         queryset = self.get_queryset()
         queryset = queryset.exclude(file="").exclude(device__test_device=True)
-        queryset = queryset.prefetch_related("org_unit")
-        queryset = queryset.prefetch_related("org_unit__version", "org_unit__version__data_source")
-        queryset = queryset.prefetch_related("org_unit__org_unit_type")
+        queryset = queryset.prefetch_related("org_unit__reference_instances")
```

Comment:
I'm a tad worried about performance issues linked to this for large exports to csv (which is something that happens a lot, since this is the mechanism used to produce dashboards out of the data)

Reply:
Looks like if you have 200 000 instances that are reference instances, we are going to fetch them all at once for this?

Reply:
For csv exports, we're streaming the content to avoid loading too much data at once in memory and this might defeat the purpose. On the other hand the django ORM is maybe smart enough to prefetch multiple times the data  needed when streaming, but I'm not sure.

Reply:
> if you have 200 000 instances that are reference instances

Is this a real life use-case?

Reply:
Anyway I moved this part explicitly out https://github.com/BLSQ/iaso/pull/807/commits/5f87799890735652ac1fa916384799ca6d62c33b